src/tensor/tensor_impls.rs

```diff
@@ -192,6 +194,20 @@ impl<S: Shape, E: Unit, D: SampleTensor<E>, T> Tensor<S, E, D, T> {
     }
 }
 
+impl<S: Shape, E: Dtype + Unit, D1: Device<E>, T, D2: Device<E>> ToDevice<D2>
```

Comment:
```suggestion
impl<S: Shape, E: Dtype + Unit, D1: DeviceStorage, T, D2: DeviceStorage> ToDevice<D2>
```

You may need to add some restrictions for vec copying, I can't remember what trait those need on top of the base DeviceStorage